core/metadata_layer.py

```diff
@@ -805,6 +805,10 @@ def mirror_link(self, model_object, data_source, link, link_obj, policy):
                     self.log.error(pool.license_exception)
             return
 
+        # If the collection is accessed later than here
+        # a TypeError: can't escape str to binary is raised
+        collection = pools[0].collection if pools else None
+
```

Comment:
This is weird.

Reply:
This one was very strange and I don't know why the TypeError was raising or why this fixes it...